### PR TITLE
feat(sentinel-transport)：make command center thread daemon

### PR DESCRIPTION
**Pull request by wutingjia**
### Describe what this PR does / why we need it
make sentinel client close resources more easier and elegant

### Does this pull request fix one issue?

#2178 
### Describe how you did it
make commad center thread daemon

### CODE REVIEW

**Comment by Anilople** [comment]
There are many place use `com.alibaba.csp.sentinel.concurrent.NamedThreadFactory.NamedThreadFactory(String)`, how about change them all to daemon in this PR?

For example:

https://github.com/alibaba/Sentinel/blob/2c12b075340ac5058dc2c57b0e272048a6ba43f6/sentinel-cluster/sentinel-cluster-client-default/src/main/java/com/alibaba/csp/sentinel/cluster/client/NettyTransportClient.java#L65

https://github.com/alibaba/Sentinel/blob/2c12b075340ac5058dc2c57b0e272048a6ba43f6/sentinel-cluster/sentinel-cluster-server-default/src/main/java/com/alibaba/csp/sentinel/cluster/flow/statistic/concurrent/expire/RegularExpireStrategy.java#L69

https://github.com/alibaba/Sentinel/blob/2c12b075340ac5058dc2c57b0e272048a6ba43f6/sentinel-extension/sentinel-datasource-nacos/src/main/java/com/alibaba/csp/sentinel/datasource/nacos/NacosDataSource.java#L50

https://github.com/alibaba/Sentinel/blob/2c12b075340ac5058dc2c57b0e272048a6ba43f6/sentinel-extension/sentinel-datasource-zookeeper/src/main/java/com/alibaba/csp/sentinel/datasource/zookeeper/ZookeeperDataSource.java#L40

https://github.com/alibaba/Sentinel/blob/2c12b075340ac5058dc2c57b0e272048a6ba43f6/sentinel-demo/sentinel-demo-dubbo/src/main/java/com/alibaba/csp/sentinel/demo/dubbo/demo2/FooConsumerBootstrap.java#L51

https://github.com/alibaba/Sentinel/blob/2c12b075340ac5058dc2c57b0e272048a6ba43f6/sentinel-transport/sentinel-transport-netty-http/src/main/java/com/alibaba/csp/sentinel/transport/command/NettyHttpCommandCenter.java#L42

**Review by linlinisme — APPROVED** [review]
LGTM!

File: sentinel-demo/sentinel-demo-dubbo/src/main/java/com/alibaba/csp/sentinel/demo/dubbo/demo2/FooConsumerBootstrap.java [hunk 1]
```diff
@@ -48,7 +48,7 @@
 
     @SuppressWarnings("PMD.ThreadPoolCreationRule")
     private static final ExecutorService pool = Executors.newFixedThreadPool(10,
-        new NamedThreadFactory("dubbo-consumer-pool"));
+        new NamedThreadFactory("dubbo-consumer-pool", true));
```
Comment (sczyh30) [inline sentinel-demo/sentinel-demo-dubbo/src/main/java/com/alibaba/csp/sentinel/demo/dubbo/demo2/FooConsumerBootstrap.java:51]: For demo maybe it's not needed (waiting may be needed)?

**Comment by sczyh30** [comment]
Thanks for contributing.

**Pull request merged**